### ISSUE TITLE
Fine-tuning GPT

### ISSUE THREAD

**Issue by dmurfet** [metauni/metauniOS#2]
We take electricity and Internet access for granted. Future generations will think the same way about advanced AIs; an environment in which AIs are *not* available and aware of what is going on, will seem broken.

In the context of metauni, this means that our AIs should be aware of seminars, what they are about, and what we have been discussing in them. Technically that means that we need an automated process of fine-tuning GPT on datasets that we generate from seminar activity. Natural sources of data:

* YouTube transcripts
* Board contents via OCR
* Texts of background reading material (books and papers) discussed in seminars

We fine-tune a single "metauni AI" on all of this, derived from GPT3 or GPT4. The dataset labels the transcripts and OCR data with what seminar it is from. For this to work a few things need to be in place:

* A way of generating transcripts (perhaps scraping YouTube to start with, or OpenAI Whisper on our own video files, running locally on the machine of whoever processes them)
* A way of associating boards with particular seminars for the purposes of data labelling (this data can be extracted from schedules which have the seminar --> pocket link in their data, but we're currently not archiving this schedule data)
* A way for seminar organisers to provide background reading material for fine-tuning (this should not involve administrative work)

In general, this should all be done in a way that involves zero to minimal administrative work. There should be some regular tasks for seminar organisers, and the rest is automated.

**Comment by dmurfet** [comment]
A link to GPT fine-tuning information: https://beta.openai.com/docs/guides/fine-tuning

**Comment by dmurfet** [comment]
Step-by-step guide to doing this : https://www.patterns.app/blog/2022/12/21/finetune-llm-tech-support
[blog.pdf](https://github.com/metauni/metauniOS/files/10297059/blog.pdf)

**Comment by dmurfet** [comment]
Having thought about this a bit more, clearly the right thing to do is "layered attention" where you have GPT at the core and then use embeddings to surface relevant material from a corpus to provide on-demand context for your core AI.

Here's a rep from OpenAI basically saying that: https://twitter.com/OfficialLoganK/status/1611084017712914450?s=20&t=rVCvmDE39KlUz9-DWISKJQ
